src/core/Akka.Streams/Stage/GraphStage.cs

```diff
@@ -1746,6 +1827,35 @@ protected internal virtual void AfterPostStop()
                 _stageActor.Stop();
                 _stageActor = null;
             }
+
+            var inProgress = AsyncCallbacksInProgress.GetAndSet(null);
+            if (inProgress is not null && inProgress.Count > 0)
```

Comment:
On stream stop, we set `AsyncCallbacksInProgress` to null to signal that we're not accepting any async callback anymore, then for each active callbacks, we signal a `StreamDetachedException` failure.